test/e2e/framework/pod/wait.go

```diff
@@ -153,41 +155,31 @@ func WaitForPodsRunningReady(ctx context.Context, c clientset.Interface, ns stri
 			Pods:                   podList.Items,
 		}, nil
 	})).WithTimeout(timeout).Should(framework.MakeMatcher(func(s *state) (func() string, error) {
-		replicas, replicaOk := int32(0), int32(0)
-		for _, rc := range s.ReplicationControllers {
-			replicas += *rc.Spec.Replicas
-			replicaOk += rc.Status.ReadyReplicas
-		}
-		for _, rs := range s.ReplicaSets {
-			replicas += *rs.Spec.Replicas
-			replicaOk += rs.Status.ReadyReplicas
-		}
 
-		nOk := int32(0)
-		notReady = int32(0)
-		failedPods := []v1.Pod{}
-		otherPods := []v1.Pod{}
-		succeededPods := []string{}
+		nOk = 0
+		badPods = []v1.Pod{}
+		otherPods = []v1.Pod{}
+		succeededPods = []string{}
 		for _, pod := range s.Pods {
 			res, err := testutils.PodRunningReady(&pod)
 			switch {
 			case res && err == nil:
 				nOk++
 			case pod.Status.Phase == v1.PodSucceeded:
-				// it doesn't make sense to wait for this pod
+				// ignore succeeded pods
 				succeededPods = append(succeededPods, pod.Name)
 			case pod.Status.Phase == v1.PodFailed:
 				// ignore failed pods that are controlled by some controller
+				// TODO either document why failures with controllers are allowed while
+				// 	failures without controllers are not, or remove this check
 				if metav1.GetControllerOf(&pod) == nil {
-					failedPods = append(failedPods, pod)
+					badPods = append(badPods, pod)
```

Comment:
This variable used to be called `badPods`, but then was changed to `failedPods`. If it doesn't actually include all `failedPods` and just the one's we've decided are especially bad, then `badPods` seems like a better name.